### PR TITLE
Nvim-rs: Fix https://github.com/Kethku/neovide/issues/86

### PR DESCRIPTION
**Pull request by KillTheMule**
Hey, 

so I finally came around trying to fix this issue for real. This is somewhat rough, but neovide builds and seems to work, so what I'd like to know at this point is: Does this fix https://github.com/Kethku/neovide/issues/86 indeed? I have no way to test, so I depend on someone else reproducing the issue and seeing if this is fixed indeed. If so, I'll work out the details in nvim-rs, and I'll send a PR for neovide when it's done.

Note: When quitting neovide with this PR, there's a panic somwhere (I guess from the `method_handler` task).  I don't think it matters at this point though :)

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -20,7 +20,7 @@ derive-new = "0.5"
 rmpv = "0.4.4"
 rust-embed = { version = "5.2.0", features = ["debug-embed"] }
 image = "0.22.3"
-nvim-rs = { git = "https://github.com/kethku/nvim-rs", features = [ "use_tokio" ] }
+nvim-rs = { git = "https://github.com/KillTheMule/nvim-rs", branch = "serial", features = [ "use_tokio" ] }
```
Comment (j4qfrost) [inline Cargo.toml:23]: Get this merged upstream first. It's bad practice to change the dependencies to another remote for maintenance reasons and security reasons.

Comment (j4qfrost) [reply Cargo.toml:23]: Just realized that you are upstream. lol

Comment (Kethku) [reply Cargo.toml:23]: I'll test it up in a bit

**Comment by Kethku** [comment]
This appears to work. I'm still testing though as the failure before was intermittent

**Comment by Kethku** [comment]
Further testing reveals that the test_read_initial_values test stalls with your branch and prints a reported exception to the console on this line: https://github.com/KillTheMule/nvim-rs/blob/serial/src/neovim.rs#L216. I think this is due to your branch, but I'm double checking.

**Comment by Kethku** [comment]
Yup definitely throws and does not throw on my hack branch. Not sure whats going on

**Comment by KillTheMule** [comment]
Can you show how you adapted the test? Maybe you're just not running the futures correctly :) I'll investigate that, but can you answer my original question if that fixes #86? The main binary does build, so you should be able to test that, right? Or maybe I've misunderstood something?

**Comment by Kethku** [comment]
It appears to work other than that. You're right I likely adapted the futures incorrectly.

I'm in the middle of a pretty large refactor of the project, so I may have to put this on pause for a bit while I settle that out. Is that ok with you?

**Comment by KillTheMule** [comment]
> I'm in the middle of a pretty large refactor of the project, so I may have to put this on pause for a bit while I settle that out. Is that ok with you?

Of course! Now that I know that this works as expected, I'm going to polish and rework stuff into something I can release. I'll try to make a PR for neovide then to switch you over to a release (crossing fingers here :))

**Comment by Kethku** [comment]
@KillTheMule I just finished the above mentioned refactor, so I'm going to reopen this PR and take a look at the changes to see what I can incorporate.

**Comment by Kethku** [comment]
Closing because I think the current workaround is sufficient.

**Pull request closed without merging**